f2py - bug in calcualtion of array dimensions

I try to include a common f77 block from legacy code.  

some calculations in the conversion go wrong.

I have in my common block lines

```
      implicit real*8 (a-h,o-z)
      parameter ( nvar  = 3       )
      parameter ( jmz    = 1983 )

      common scmt(3*(jmz+1)), dbuf(nvar*(nvar+1) * (jmz+2))
```

resulting an a comlilation error

```
/tmp/tmpcpniavle/src.linux-x86_64-3.4/_keplermodule.c:1386:22: error: ânvarâ undeclared here (not in a function)
   {"dbuf",1,{{5955 * nvar + 5955}},NPY_DOUBLE},
```
it seems a bad line is generated and f2py does not resolve multiple occurrences (nvar) of the same parameter/variable in an expression?  at least inside sufficiently complex expressions?

If I define a parameter with nvar*(nvar+1) and use that new parameter, things work, but maybe if you get a chance to fix this fluke as well, that would be nice.  (or let me me know in what function for crackfortran even to look...)

--- a/numpy/f2py/crackfortran.py
+++ b/numpy/f2py/crackfortran.py
@@ -2624,11 +2624,12 @@ def analyzevars(block):
                     if d in params:
                         d = str(params[d])
                     for p in list(params.keys()):
-                        m = re.match(
-                            r'(?P<before>.*?)\b' + p + r'\b(?P<after>.*)', d, re.I)
-                        if m:
+                        re_1 = re.compile(r'(?P<before>.*?)\b' + p + r'\b(?P<after>.*)', re.I)
+                        m = re_1.match(d)
+                        while m:
                             d = m.group('before') + \
                                 str(params[p]) + m.group('after')
+                            m = re_1.match(d)
                     if d == star:
                         dl = [star]
                     else:

